src/System.IO.Pipelines/System/IO/Pipelines/ReadCursor.cs

```diff
@@ -211,15 +269,15 @@ private bool TryGetBufferMultiBlock(ReadCursor end, out Memory<byte> data)
 
             while (true)
             {
-                var wasLastSegment = segment.Next == null || end.Segment == segment;
+                var wasLastSegment = segment.Next == null || end == segment;
```

Comment:
if `segment.Next == null`, doesn't that mean end didn't follow start? Why not throw?

Reply:
This logic needs to be merged with `MemoryEnumerator.MoveNext`